skbio/metadata/tests/test_interval.py

```diff
@@ -446,6 +446,34 @@ def test_init_upper_bound_lt_lower_bound(self):
         with self.assertRaises(ValueError):
             IntervalMetadata(-1)
 
+    def test_upper_bound_is_none(self):
```

Comment:
...unless you are ok not to check `upper/lower_bound` in `__eq__`, which probably makes more sense.